src/libs/viewpoints/viewpoint_extractor.ts

```diff
@@ -0,0 +1,536 @@
+    const viewpoint = new Viewpoint('Non Sortal View');
+
+    /**Step 1 Get all Non-Sortals in model
+     * 
+     * Step 2 Get all non-sortals decendants 
+     * 
+     * Step 3 Get all decendants sortal ancestors 
+     * 
+     * Step 4 Get all relations between all classes 
+     * 
+     */
+    let classes = this.project.getAllClassesByStereotype(ClassStereotype.ROLE_MIXIN);
+
+    classes = classes.concat(this.project.getAllClassesByStereotype(ClassStereotype.MIXIN));
+    classes = classes.concat(this.project.getAllClassesByStereotype(ClassStereotype.CATEGORY));
+    classes = classes.concat(this.project.getAllClassesByStereotype(ClassStereotype.PHASE_MIXIN));
+
+    let decendentes = [];
+    for (let i = 0; i < classes.length; i++) {
+      decendentes = decendentes.concat(classes[i].getDescendants());
+    }
+
+    let generals = [];
+    for (let i = 0; i < decendentes.length; i++) {
+      generals = generals.concat(decendentes[i].getAncestors());
+    }
+
+    classes = classes.concat(generals);
+
+    classes = classes.concat(decendentes);
+
+    viewpoint.addClasses(classes);
+    viewpoint.removeDuplicates();
+
+
+    viewpoint.addRelations(this.getRelationsBetweenClasses(viewpoint.classes));
+    viewpoint.addGeneralizations(this.getGeneralizationsBetweenClasses(viewpoint.classes));
+
+    viewpoint.removeDuplicates();
+
+    return viewpoint;
+  }
+
+  /**
+     * This method filter all Modes, their descendants, then the decendants sortal ancestors  and related relations and group them in a view
+     * 
+     * @returns A viewpoint extracted using the Mode Pattern
+     */
+  modeView(): Viewpoint {
+    const viewpoint = new Viewpoint('Mode View');
+
+    /**Step 1 Get all Modes in model
+      * 
+      * Step 2 Get all modes decendants 
+      * 
+      * Step 3 Get all decendants sortal ancestors 
+      * 
+      * Step 4 Get all relations between all classes
+      * 
+      */
+    let classes = this.project.getAllClassesByStereotype(ClassStereotype.MODE);
+
+    let decendentes = [];
+    for (let i = 0; i < classes.length; i++) {
+      decendentes = decendentes.concat(classes[i].getDescendants());
+    }
+
+    let generals = [];
+    for (let i = 0; i < decendentes.length; i++) {
+      generals = generals.concat(decendentes[i].getAncestors());
+    }
+
+    classes = classes.concat(generals);
+
+    classes = classes.concat(decendentes);
+
+    viewpoint.addClasses(classes);
+
+    viewpoint.removeDuplicates();
+
+
+    viewpoint.addRelations(this.getRelationsBetweenClasses(viewpoint.classes));
+    viewpoint.addGeneralizations(this.getGeneralizationsBetweenClasses(viewpoint.classes));
+
+    viewpoint.removeDuplicates();
+
+    return viewpoint;
+  }
+
+  /**
+   * Verify is a class participates in relation
+   * @param _class The classe to be verified
+   * @param relation The relation where the class may be participating
+   */
+  isClassInRelation(_class: Class, relation: Relation): boolean {
```

Comment:
This should be a method of the `Relation` class named `involves()`

Reply:
Done